app/features/receive/use-receive-cashu-token.tsx

```diff
@@ -0,0 +1,358 @@
+import { CashuMint, CashuWallet, type Token } from '@cashu/cashu-ts';
+import { useQuery } from '@tanstack/react-query';
+import { useState } from 'react';
+import type { Account, CashuAccount } from '~/features/accounts/account';
+import {
+  useAccounts,
+  useAddAccount,
+  useDefaultAccount,
+} from '~/features/accounts/account-hooks';
+import { tokenToMoney } from '~/features/shared/cashu';
+import { useToast } from '~/hooks/use-toast';
+import {
+  getP2PKPubkeyFromProofs,
+  getUnspentProofsFromToken,
+  isP2PKSecret,
+  isPlainSecret,
+  swapProofsToWallet,
+} from '~/lib/cashu';
+import { checkIsTestMint, getMintInfo } from '~/lib/cashu';
+import type { MintInfo } from '~/lib/cashu';
+import type { Money } from '~/lib/money';
+import { useNavigateWithViewTransition } from '~/lib/transitions';
+import { getDefaultUnit } from '../shared/currencies';
+
+type UseReceiveCashuTokenProps = {
+  token: Token;
+  cashuPubKey: string;
+};
+
+type UseReceiveCashuTokenData = {
+  /** The token with only claimable proofs or the full token if it cannot be claimed */
+  displayToken: Token;
+  /** Whether the token can be claimed (fully or partially) by this user */
+  canClaim: boolean;
+  /** The account to receive the token */
+  receiveAccount: CashuAccount;
+  /** The account that the token is from */
+  sourceAccount: CashuAccount;
+  /** True if the source account cannot make a lightning payment to other accounts */
+  disableCrossMintSwap: boolean;
+  /** Whether the selected account is the source mint */
+  selectedAccountIsSource: boolean;
+  /** The reason why the user cannot claim the token */
+  cannotClaimReason: string | undefined;
+  /** The accounts that the user can select to receive the token */
+  selectableAccounts: CashuAccount[];
+  /** The token as Money object */
+  money: Money;
+  /**
+   * Whether the token's mint is in the user's accounts. Will be undefined
+   * if the user is not initialized (ie. they clicked a token link but do not have an account)
+   */
+  isMintKnown: boolean | undefined;
```

Comment:
would isMintTrusted be better name?

also what do you mean by `if the user is not initialized (ie. they clicked a token link but do not have an account)`?

if user is not initilized the app should never get past the protected route

Reply:
lets handle public page later and not have this hook worry about that now